### PR TITLE
Move integration tests to miden crate

### PR DESCRIPTION
**Pull request by grjte**
This PR addresses #144 and moves all integration tests from the processor to the integration `tests` directory in the `miden` crate.

It also refactors the auxiliary table trace into a module and refactors its tests from integration tests to unit tests.

### CODE REVIEW

File: processor/src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@ mod operations;
 
 mod system;
 use system::System;
+pub use system::FMP_MIN;
```
Comment (bobbinth) [inline processor/src/lib.rs:17]: We should move this constant into the `core` create - but probably in another PR.

File: processor/src/aux_table/mod.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            .try_into()
+            .expect("failed to convert vector to array");
+
+        Self { trace }
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns an execution trace of the top 16 stack slots and helper columns as a single array.
```
Comment (bobbinth) [inline processor/src/aux_table/mod.rs:70]: Seems like a wrong comment got copied over here.

**Review by bobbinth — APPROVED** [review]
Looks great! Thank you! Just one doc-related comment inline.

File: processor/src/lib.rs (hunk 1 above)
Comment (grjte) [reply processor/src/lib.rs:17]: Sure. Before refactoring the aux table there were quite a few constants needed for tests that I didn't feel really belonged in core (Hasher & Bitwise internals), so I ended up not moving any of them over. I agree I could see this one going over, but I'll leave it for the future.

File: processor/src/aux_table/mod.rs (hunk 2 above)
Comment (grjte) [reply processor/src/aux_table/mod.rs]: Oops, not sure how that happened. Fixed now!

**Comment by grjte** [comment]
@bobbinth addressed comments & rebased to migrate the latest tests as well

**Pull request merged**